documentation/Spring-native-vs-Quarkus.asciidoc

```diff
@@ -0,0 +1,102 @@
+====
+
+Spring Native provides support for compiling Spring applications to
+native executables using the GraalVM native-image compiler. Using a
+native image provides some key advantages, such as instant startup,
+instant peak performance, and reduced memory consumption. However, there are
+also some drawbacks: Creating a native image is a heavy process that is
+slower than a regular application. A native image also has fewer runtime
+optimizations after its warmup. Furthermore, it is less mature than the
+JVM and comes with some different behaviors.
+
+*Difference in regular JVM vs GraalVM*      
+
+* Static analysis of the application from the main entry point is
+performed at build time.
+* Unused parts are removed at build time.
+* Configuration required for reflection, resources, and dynamic proxies.
+* Classpath is fixed at build time.
+* No class lazy loading: everything shipped in the executables will be
+loaded in memory on startup.
+* Some code will run at build time.
+
+There are https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/Limitations.md[limitations] around some aspects of Java applications that are not fully supported and therefore should be made aware of before using Spring Native.
+
+=== Build time and start time for apps
+
+[cols=",,",options="header",]
+|===
+|Framework |build time |start time
+|Spring Native |19.615s |2.913s
+|Quarkus Native executable |52.818s |0.802s
+|===
+
+=== Memory footprints
+
+[cols=",",options="header",]
+|===
+|Framework |memory footprint
+|Spring Native |109 MB
+|Quarkus Native executable |75 MB
+|===
+
+=== Considering devonfw best practices
+
+devonfw actively supports Spring, as it is widely used and is unmatched regarding stability and support, despite being slower in build and start times and having a bigger memory footprint in general. 
+While Spring Native solves these problems quite well, it is still in beta and therefore not considered stable enough to be an option for devonfw as of now. 
+devonfw Spring modules could be supported in a Spring Native application, but there is no guarantee that it will work.
+Despite being quite new, we still consider Quarkus as stable and mature enough to be supported, even though there are no Quarkus modules in devonfw at the moment.
+
+=== General recommendations and conclusion
+
+For Greenfield projects which are mainly focusing on microservices or the cloud, Quarkus can be recommended, as it is more suitable for their requirements. 
+Spring projects, should be able to use Spring Native in the future in order to improve their performance. 
+We also provide a link:quarkus/getting-started-for-spring-developers.asciidoc[guide
+for Spring developers] who want to adopt or try Quarkus for their
+(next) projects as it really has some gamechanging advantages over
+Spring.
+As of now, Quarkus and Spring Native both have their own use cases, but neither of them can be considered as a general recommendation for everybody.
```

Comment:
Let's go more strict.

Quarkus and Spring Native both have their own use cases. Under the consideration of the limitations of GraalVM (SubstrateVM) to be used for native images built by Quarkus and Spring Native, there is a strong recommendation towards Quarkus from devonfw. One essential differentiation has to be made on the decision for native or against native applications - the foreseen performance optimization of the JIT compiler of the JVM, which is not available anymore in a native image deployment. For sure, both component frameworks will also run on a JVM getting advantage again from JIT compilation, but depending on the overall landscape then, it is recommended to stay with the knowledge of the available teams, e.g. continue making use of devon4j based on spring or even if already in that state also here make use of Quarkus on JVM.